addon/webextension/background/analytics.js

```diff
@@ -0,0 +1,74 @@
+/* globals main, auth, catcher, deviceInfo, communication */
+
+window.analytics = (function () {
+  let exports = {};
+
+  let telemetryPrefKnown = false;
+  let telemetryPref;
+
+  exports.sendEvent = function (action, label, options) {
+    let eventCategory = "addon";
+    if (! telemetryPrefKnown) {
+      console.warn("sendEvent called before we were able to refresh");
+      return Promise.resolve();
+    }
+    if (! telemetryPref) {
+      console.info(`Cancelled sendEvent ${eventCategory}/${action}/${label || 'none'} ${JSON.stringify(options)}`);
```

Comment:
What's a good way to handle a pref in a webextension?  In the SDK version we used about:config prefs for some things like this, but there doesn't seem to be a webextension equivalent, so either I'd need a whole options page (which would be bad since these aren't options that are meaningful to users), or build the pref into the source itself.  Or hopefully there's a better way I'm not aware of...?